src/main/java/de/tum/in/www1/artemis/repository/ProgrammingExerciseRepository.java

```diff
@@ -57,6 +57,16 @@
     @Query("select distinct pe from ProgrammingExercise pe left join fetch pe.studentParticipations")
     List<ProgrammingExercise> findAllWithEagerParticipations();
 
+
+    // Get all programming exercises that need to be scheduled: Those must satisfy one of the following requirements:
+    // 1. The release date is in the future --> Schedule combine template commits
+    // 2. The build and test student submissions after deadline date is in the future
+    // 3. Manual assessment is enabled and the due date is in the future
+    @Query("select distinct pe from ProgrammingExercise pe where pe.releaseDate > :#{#now} or " +
+        "pe.buildAndTestStudentSubmissionsAfterDueDate > :#{#now} or " +
+        "pe.assessmentType <> 'AUTOMATIC' and pe.dueDate > :#{#now}")
```

Comment:
```suggestion
    @Query("""
         select distinct pe from ProgrammingExercise pe where pe.releaseDate > :#{#now} or
         pe.buildAndTestStudentSubmissionsAfterDueDate > :#{#now} or
         pe.assessmentType <> 'AUTOMATIC' and pe.dueDate > :#{#now}
             """)
```
please use multi-line Strings in your queries :D